packages/form-plugin/src/directive.ts

```diff
@@ -27,102 +38,95 @@ export class FormDirective implements OnInit, OnDestroy {
+      if (this._updating || !model) {
+        return;
+      }
+
+      this.form.patchValue(model);
+      this._cd.markForCheck();
+    });
 
     // On first state change, sync form model, status and dirty with state
     this._store
-      .select(state => getValue(state, `${this.path}`))
-      .pipe(
-        takeUntil(this._destroy$),
-        first()
-      )
-      .subscribe(state => {
+      .selectOnce(state => getValue(state, this.path))
+      .subscribe(() => {
         this._store.dispatch([
           new UpdateFormValue({
             path: this.path,
-            value: this._formGroupDirective.form.getRawValue()
+            value: this.form.getRawValue()
           }),
           new UpdateFormStatus({
             path: this.path,
-            status: this._formGroupDirective.form.status
+            status: this.form.status
           }),
           new UpdateFormDirty({
             path: this.path,
-            dirty: this._formGroupDirective.form.dirty
+            dirty: this.form.dirty
           })
         ]);
       });
 
-    this._store
-      .select(state => getValue(state, `${this.path}.dirty`))
-      .pipe(takeUntil(this._destroy$))
-      .subscribe(dirty => {
-        if (this._formGroupDirective.form.dirty !== dirty) {
-          if (dirty === true) {
-            this._formGroupDirective.form.markAsDirty();
-            this._cd.markForCheck();
-          } else if (dirty === false) {
-            this._formGroupDirective.form.markAsPristine();
-            this._cd.markForCheck();
-          }
-        }
-      });
+    this.getStateStream(`${this.path}.dirty`).subscribe(dirty => {
+      if (this.form.dirty === dirty || typeof dirty !== 'boolean') {
+        return;
+      }
 
-    this._store
-      .select(state => getValue(state, `${this.path}.disabled`))
-      .pipe(takeUntil(this._destroy$))
-      .subscribe(disabled => {
-        if (this._formGroupDirective.form.disabled !== disabled) {
-          if (disabled === true) {
-            this._formGroupDirective.form.disable();
-            this._cd.markForCheck();
-          } else if (disabled === false) {
-            this._formGroupDirective.form.enable();
-            this._cd.markForCheck();
-          }
-        }
-      });
+      if (dirty) {
+        this.form.markAsDirty();
+      } else {
+        this.form.markAsPristine();
+      }
+
+      this._cd.markForCheck();
+    });
+
+    this.getStateStream(`${this.path}.disabled`).subscribe(disabled => {
+      if (this.form.disabled === disabled || typeof disabled !== 'boolean') {
+        return;
+      }
+
+      if (disabled) {
+        this.form.disable();
+      } else {
+        this.form.enable();
+      }
+
+      this._cd.markForCheck();
+    });
 
     this._formGroupDirective
       .valueChanges!.pipe(
         debounceTime(this.debounce),
+        distinctUntilChanged((a, b) => JSON.stringify(a) === JSON.stringify(b)),
```

Comment:
This looks like a bug fix as opposed to a refactor. Could this be in a separate PR?